figtree.go

```diff
@@ -1388,6 +1401,9 @@ func (m *Merger) mergeStructs(dst reflect.Value, src mergeSource, overwrite bool
 				return assignErr
 			}
 			changed = changed || fieldChanged
+			if fieldChanged {
+				return nil
```

Comment:
This short-circuits the fallthrough for the case where the assignment actually happened, we don't need to try to look for different structures since the assignment succeeded.  This is necessary for the structure-with-UnmarhsalYAML case.